### PR TITLE
Noos project

### PR DESCRIPTION
**Pull request by mchindri**
- Create a project that builds noos as a library
- Remove no_os_test_project
- Create no_os_blinky_example form no_os_test_project source and link to no_os_library_project

Why this modification are useful is explained in the first commit.
This PR will be set as draft until some problems related with noos repo are solved:
- [ ] What to do with (sd.h and fatfs.h ...)
- [ ] Modify uart_extra.h to avoid including bsp in main app
- [ ] Can we save somehow debug an run configurations?
-- If yes:
- [ ] -- Find how to reset from button and get back to code breakpoint. 
- [ ] -- Create Debug : (use debug build and main breakpoint)
- [ ] -- Create Run: (use release build without main breakpoint)

### CODE REVIEW

**Comment by adrimbarean** [comment]
Second commit needs signature.

File: projects/no-os_library_project/RTE/RTE_Components.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+*/
+#ifndef RTE_COMPONENTS_H
+#define RTE_COMPONENTS_H
+
+/*
+ * Define the Device Header File:
+*/
+#define CMSIS_device_header "ADuCM3029.h"
+
+
```
Comment (amiclaus) [inline projects/no-os_library_project/RTE/RTE_Components.h:16]: nitpick: extra line

Comment (mchindri) [reply projects/no-os_library_project/RTE/RTE_Components.h:16]: This file is generated when creating the project. Modifications shouldn't be done in it

Comment (amiclaus) [reply projects/no-os_library_project/RTE/RTE_Components.h:16]: Okey then

File: projects/no-os_library_project/.project [hunk 2]
```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mirceacaprioru) [inline projects/no-os_library_project/.project:1]: commit description should be limited to 75 character/line

Comment (mirceacaprioru) [reply projects/no-os_library_project/.project:1]: also there are 2 typos: projecte -> project

File: projects/ADuCM3029_no_os_blinky_example/src/blinky_test.c [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*****************************************************************************
+ * ADuCM3029_no_os_test_project.c
+ *****************************************************************************/
+
+#include <sys/platform.h>
```
Comment (mirceacaprioru) [inline projects/ADuCM3029_no_os_blinky_example/src/blinky_test.c:5]: it seems that blinky_test and blinky_example are the same thing. Remove one of them.

File: projects/ADuCM3029_no_os_test_project/.cproject [hunk 4]
```diff
@@ -1,398 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (mirceacaprioru) [inline projects/ADuCM3029_no_os_test_project/.cproject:1]: these generates a lot of noise to remove a project and add it again. Why not just modify this one ?

**Pull request closed without merging**